lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -296,6 +297,16 @@ class ReportUIFeatures {
     });
   }
 
+  _setupElementScreenshotOverlay() {
+    const json = this.json;
+    const fullPageScreenshot = /** @type {LH.Artifacts.FullPageScreenshot | undefined} */ (
+      json.audits['full-page-screenshot'] && json.audits['full-page-screenshot'].details);
+    if (!fullPageScreenshot) return;
+
+    ElementScreenshotRenderer.installOverlayFeature(
```

Comment:
the shared requirements here between report-ui-features feels like an actual good reason to have this thing be an instance that can check if it's been initialized and opt out of the rendering

Reply:
I tried but I'm not sure it'll work .... https://github.com/GoogleChrome/lighthouse/commit/e65ea6d685d3b66c5027511fe330f6cbfab637dc#diff-67b57a84752abea0be625494faa645fcL522

the pattern used by `CriticalRequestChainRenderer.render` makes it simpler imo. I can't figure out how to retain a single instance of thing and use it both for the installing overlay (an optional ui feature that we may disable in some channels) and augmenting `node` detail types in details renderer.

Reply:
wfm